WrightTools/artists/_base.py

```diff
@@ -332,50 +332,28 @@ def legend(self, *args, **kwargs):
             kwargs["framealpha"] = 1.
         return super().legend(*args, **kwargs)
 
-    def pcolor(self, *args, **kwargs):
-        """Create a pseudocolor plot of a 2-D array.
-
-        Uses pcolor_helper to ensure that color boundaries are drawn
-        bisecting point positions, when possible.
-
-        Parameters
-        ----------
-        data : 2D WrightTools.data.Data object
-            Data to plot.
-        channel : int or string (optional)
-            Channel index or name. Default is 0.
-        dynamic_range : boolean (optional)
-            Force plotting of all contours, overloading for major extent. Only applies to signed
-            data. Default is False.
-        autolabel : {'none', 'both', 'x', 'y'}  (optional)
-            Parameterize application of labels directly from data object. Default is none.
-        xlabel : string (optional)
-            xlabel. Default is None.
-        ylabel : string (optional)
-            ylabel. Default is None.
-        **kwargs
-            matplotlib.axes.Axes.pcolor__ optional keyword arguments.
-
-            __ https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.pcolor.html
-
-        Returns
-        -------
-        matplotlib.collections.PolyCollection
-        """
+    def _parse_plot_args(self, *args, **kwargs):
+        plot_type = kwargs.pop("plot_type", False)
         args = list(args)  # offer pop, append etc
         channel = kwargs.pop("channel", 0)
         dynamic_range = kwargs.pop("dynamic_range", False)
-        # unpack data object, if given
         if isinstance(args[0], Data):
             data = args.pop(0)
-            if not data.ndim == 2:
+            if plot_type in ['pcolor', 'pcolormesh', 'contour', 'contourf']:
+                ndim = 2
+            elif plot_type in ['plot']:
```

Comment:
Is this intended to be used for `plot`?
If so, there are some parts of the method (e.g. lines 350 - 360 or so) which assume 2D axes

Reply:
@p770193 (just bumping because it had gotten collapsed)